core/src/main/java/com/graphhopper/GraphHopper.java

```diff
@@ -1048,18 +1054,13 @@ protected void loadOrPrepareLM(boolean closeEarly) {
         }
 
         if (landmarkSplittingFeatureCollection != null && !landmarkSplittingFeatureCollection.getFeatures().isEmpty()) {
-            SpatialRuleLookup ruleLookup = SpatialRuleLookupBuilder.buildIndex(
-                    Collections.singletonList(landmarkSplittingFeatureCollection), "area",
-                    (id, polygons) -> new AbstractSpatialRule(polygons) {
-                        @Override
-                        public String getId() {
-                            return id;
-                        }
-                    });
+            List<CustomArea> landmarkAreas = CustomAreaHelper.loadAreas(landmarkSplittingFeatureCollection, "area");
+            CustomAreaLookup landmarkAreaLookup = landmarkAreas.isEmpty() ? CustomAreaLookup.EMPTY : new CustomAreaLookupJTS(landmarkAreas);
+
             for (PrepareLandmarks prep : getLMPreparationHandler().getPreparations()) {
                 // the ruleLookup splits certain areas from each other but avoids making this a permanent change so that other algorithms still can route through these regions.
-                if (ruleLookup != null && !ruleLookup.getRules().isEmpty()) {
-                    prep.setSpatialRuleLookup(ruleLookup);
+                if (landmarkAreaLookup != CustomAreaLookup.EMPTY) {
+                    prep.setCustomAreaLookup(landmarkAreaLookup);
                 }
             }
```

Comment:
IMO it would be a bit more clear if we do
```java
if(!landmarkAreas.isEmpty()) {
  CustomAreaLookup landmarkAreaLookup = new CustomAreaLookupJTS(landmarkAreas);
  for (PrepareLandmarks prep : getLMPreparationHandler().getPreparations()) {
     prep.setCustomAreaLookup(landmarkAreaLookup); 
  }
}
```
Or did I overlooked something?